yorkie/src/main/kotlin/dev/yorkie/document/json/JsonTree.kt

```diff
@@ -34,6 +37,218 @@ public class JsonTree internal constructor(
+     */
+    private fun separateSplit(
+        treePos: TreePos<CrdtTreeNode>,
+        path: List<Int>,
+    ): List<Triple<List<Int>, List<Int>, TreeNode?>> {
+        val node = treePos.node
+        val parentPath = path.dropLast(1)
+        val parentNode = node.parent
+        val last = if (node.isText && parentNode != null) {
+            parentNode.getChildrenText().length
+        } else {
+            node.children.size
+        }
+        val toPath = parentPath + last
+        val insertPath = parentPath.toMutableList()
+        insertPath[insertPath.size - 1] = insertPath.last() + 1
+
+        val res = mutableListOf<Triple<List<Int>, List<Int>, TreeNode?>>()
+        if (path != toPath) {
+            res.add(
+                Triple(
+                    first = path,
+                    second = toPath,
+                    third = null,
+                ),
+            )
+        }
+
+        val newNode = createSplitNode(node, path.last())
+        res.add(
+            Triple(
+                first = insertPath,
+                second = insertPath,
+                third = newNode,
+            ),
+        )
+
+        return res
+    }
+
+    /**
+     * `separateMerge` separates the merge operation into insert and delete operations.
+     */
+    private fun separateMerge(
+        treePos: TreePos<CrdtTreeNode>,
+        path: List<Int>,
+    ): List<Triple<List<Int>, List<Int>, Array<TreeNode>>> {
+        val parentNode = treePos.node
+        val offset = treePos.offset
+        val node = parentNode.children[offset]
+        val leftSiblingNode = parentNode.children[offset - 1]
+        val children = node.children
+        val parentPath = path.dropLast(1)
+        val res = mutableListOf<Triple<List<Int>, List<Int>, Array<TreeNode>>>()
+
+        // Add initial fromPath -> toPath mapping
+        res.add(
+            Triple(
+                first = path.toList(),
+                second = parentPath + (offset + 1),
+                third = emptyArray(),
+            ),
+        )
+
+        // If no children, return early
+        if (children.isEmpty()) {
+            return res
+        }
+
+        // Determine insertPath
+        val insertPath = buildList {
+            addAll(parentPath)
+            add(offset - 1)
+            val length = if (leftSiblingNode.hasTextChild) {
+                leftSiblingNode.getChildrenText().length
+            } else {
+                leftSiblingNode.children.size
+            }
+            add(length)
+        }
+
+        // Convert children to TreeNode
+        val nodes = children.map {
+            it.toTreeNode().toJsonTreeNode()
+        }.toTypedArray()
+
+        res.add(
+            Triple(
+                first = insertPath,
+                second = insertPath,
+                third = nodes,
+            ),
+        )
+
+        return res
+    }
+
+    /**
+     * `splitByPath` splits the tree by the given path.
```

Comment:
According to [documentation-comments on Kotlin](https://kotlinlang.org/docs/coding-conventions.html#documentation-comments)

```suggestion
     * `splitByPath` splits the tree by the given [path].
```